cli/running/running.go

```diff
@@ -508,17 +510,24 @@ func FillCtx(cliOpts *config.CliOpts, cmdCtx *cmdcontext.CmdCtx,
 
 // Start an Instance.
 func Start(cmdCtx *cmdcontext.CmdCtx, run *InstanceCtx) error {
+	sigCtx, cancel := signal.NotifyContext(context.Background())
 	if err := process_utils.CreatePIDFile(run.PIDFile); err != nil {
 		return err
 	}
 
-	defer cleanup(cmdCtx, run)
+	defer cleanup(run)
 
 	logger := createLogger(run)
 	provider := providerImpl{cmdCtx: cmdCtx, instanceCtx: run}
 	wd := NewWatchdog(run.Restartable, 5*time.Second, logger, &provider)
-	wd.Start()
 
+	select {
+	case <-sigCtx.Done():
+		return nil
+	default:
+		cancel()
+		wd.Start()
```

Comment:
Moved a `Reset` logic to the logic of signal handling loop. It will be called right before `Notify` call. An alternative way is to use some single signal handling loop and the logic of replacing the callbacks for interrupt signals.